src/Datalust.Piggy/Database/DatabaseConnector.cs

```diff
@@ -1,98 +1,96 @@
-        {
-            if (connectionString == null) throw new ArgumentNullException(nameof(connectionString));
+    /// <summary>
+    /// Connect to the specified database.
+    /// </summary>
+    /// <param name="connectionString">A connection string identifying the database.</param>
+    /// <returns>An open database connection.</returns>
+    public static NpgsqlConnection Connect(string connectionString)
+    {
+        if (connectionString == null) throw new ArgumentNullException(nameof(connectionString));
 
-            var conn = new NpgsqlConnection(connectionString);
+        var conn = new NpgsqlConnection(connectionString);
 
-            var host = conn.Host;
-            if (conn.Host == null && !string.IsNullOrWhiteSpace(conn.ConnectionString))
-            {
-                var parts = ConnectionStringParser.Parse(conn.ConnectionString);
-                if (parts.TryGetValue("Host", out var value)) host = value;
-            }
+        var host = conn.Host;
+        if (conn.Host == null && !string.IsNullOrWhiteSpace(conn.ConnectionString))
+        {
+            var parts = ConnectionStringParser.Parse(conn.ConnectionString);
+            if (parts.TryGetValue("Host", out var value)) host = value;
+        }
 
-            Log.Information("Connecting to database {Database} on {Host}", conn.Database, host);
+        Log.Information("Connecting to database {Database} on {Host}", conn.Database, host);
 
-            try
-            {
-                conn.Open();
-            }
-            catch
-            {
-                conn.Dispose();
-                throw;
-            }
+        try
+        {
+            conn.Open();
+        }
+        catch
+        {
+            conn.Dispose();
+            throw;
+        }
 
-            Log.Information("Connected");
+        Log.Information("Connected");
 
-            return conn;
-        }
+        return conn;
+    }
 
-        static bool TryCreate(string host, string database, string username, string password)
+    static bool TryCreate(string host, string database, string username, string password)
+    {
+        Log.Information("Database does not exist; attempting to create it");
+
+        var postgresCstr = $"Host={host};Username={username};Password={password};Database=postgres";
+        using (var postgresConn = new NpgsqlConnection(postgresCstr))
         {
-            Log.Information("Database does not exist; attempting to create it");
+            postgresConn.Open();
 
-            var postgresCstr = $"Host={host};Username={username};Password={password};Database=postgres";
-            using (var postgresConn = new NpgsqlConnection(postgresCstr))
+            Log.Information("Creating database {Database} on {Host} with owner {Owner} and default options", database, host, username);
+            using (var createCommand = new NpgsqlCommand($"CREATE DATABASE {database} WITH OWNER = {username} ENCODING = 'UTF8' CONNECTION LIMIT = -1;", postgresConn))
```

Comment:
Can / should this be parameterized?

Reply:
Worth looking into again; IIRC this was not possible last time around.